app/controllers/appointments_controller.rb

```diff
@@ -1,31 +1,43 @@
+# frozen_string_literal: true
+
 class AppointmentsController < ApplicationController
   include Authenticatable
   before_action :set_user
-  before_action :set_appointment, only: [:show, :edit, :update, :destroy]
+  before_action :set_appointment, only: %i[show edit update destroy]
 
   def index
     @appointments = @user.appointments
   end
 
-  def show
-  end
+  def show; end
 
   def new
     @appointment = @user.appointments.build
+    if params[:day].present?
+      @date = params[:day]
+      @date = Date.parse(@date)
+      start_of_day = @date.beginning_of_day
+      end_of_day = @date.end_of_day
+
+      @time_slots = Availability.where('start_time >= ? AND end_time <= ?', start_of_day, end_of_day)
+    else
+      @time_slots = []
+    end
   end
 
+  def edit; end
+
   def create
     @appointment = @user.appointments.build(appointment_params)
+    @appointment.user_id = current_user.id
```

Comment:
The appointment belongs to the user. when you create an apointment you also create the user_id who created the appointment. I don't think you need `@appointment.user_id = current_user.id`